### DISCUSSION TITLE
Programmatically usage

### DISCUSSION THREAD

**Discussion by konradmichalik** [deployphp/deployer#2379] [General]
Hello there, i can't find an answer by my own so far, maybe the question has already been answered.
Is it possible to use deployer programmatically within the php context? So e.g. hosts configuration can be done dynamically?
I want to use deployer as part of a dynamic multi host php application.

**Comment by antonmedv** [comment]
Yes, it's just a PHP code, you can use it as you want to.

**Reply by konradmichalik** [reply]
How can I start the "deploy" task? Is there something like a run command for starting tasks? 

I mean from another PHP Class. Like the following:
```
<?php
...
host(...)

task('deploy',[...]);

$deployer->run();
```

**Reply by antonmedv** [reply]
$deployer->run();

**Reply by konradmichalik** [reply]
Can you please specify this a little bit more? Is there anywhere a working example for this?

When I try to run the following line, I need to specify a deploy file.  How is the correct syntax for this?

```
\Deployer\Deployer::run('example-stage'); 
```

**Reply by konradmichalik** [reply]
And also the question if its possible to log the single steps?

**Reply by konradmichalik** [reply]
Hello @antonmedv,
do you have a hint, how to implement a programmatically based deployment?

**Reply by antonmedv** [reply]
Use Deployer instance.

**Reply by konradmichalik** [reply]
Could you please provide a working example for this scenario? I only see that a external deploy file is needed to run the deployment and so I don't understand to programmatically define all host and task information.

**Reply by antonmedv** [reply]
Here is the simplest example for programmatically invoking Deployer from an PHP script:
```php
<?php
// Execute deploy command.
$output = null;
$status = null;
exec('php deployer.phar deploy', $output, $status);
echo "Returned with status $status and output:\n";
print_r($output);
?>
```

**Reply by konradmichalik** [reply]
Okay, I thought so, there is just the posibility to run the deployer as command tool and not to use deployer as PHP service within another PHP application.
